test/variational/test_variational.py

```diff
@@ -105,6 +105,14 @@ def vstate(request):
 
     vs = nk.vqs.MCState(sa, ma, n_samples=1000, seed=SEED)
 
+    # wrap the apply function with a check that the samples have exactly 0 or 1 batch axis
+    def apply_fun_check_shape(apply_fun, params, x, *args, **kwargs):
+        assert x.ndim in (1, 2)
+        return apply_fun(params, x, *args, **kwargs)
+
+    apply_fun = vs._apply_fun
+    vs._apply_fun = partial(apply_fun_check_shape, apply_fun)
+
```

Comment:
let me know if this is too hacky.
I felt like it would be good to check it in many places, and here we can do so for free (compared to adding a single dedicated test)

Also, if you check for ndim==2 there is quite a few places where the ndim=1, but apparently we always wrap with to support this  https://github.com/netket/netket/blob/fe7629b996250b0661e78fb5f79e83648d1bc3c3/netket/vqs/mc/mc_state/state.py#L197-L205